src/main/java/org/jabref/logic/citationkeypattern/AbstractCitationKeyPattern.java

```diff
@@ -33,12 +32,19 @@ public static List<String> split(String bibtexKeyPattern) {
         // A holder for fields of the entry to be used for the key
         List<String> fieldList = new ArrayList<>();
 
-        // Before we do anything, we add the parameter to the ArrayLIst
+        // Before we do anything, we add the parameter to the ArrayList
         fieldList.add(bibtexKeyPattern);
 
-        StringTokenizer tok = new StringTokenizer(bibtexKeyPattern, "[]", true);
-        while (tok.hasMoreTokens()) {
-            fieldList.add(tok.nextToken());
+        // regex to split around a character
+        String WITH_DELIMITER = "((?<=%1$s)|(?=%1$s))";
```

Comment:
Outdated comment - since we keep `StringTokenizer`.

This regex reads a bit complicated. Can you add a comment that this regex is put like this to keep the character in the resulting split?

I needed to consult ChatGPT to understand ^^

![image](https://user-images.githubusercontent.com/1366654/229436252-7e684f08-67e8-4276-8107-5632d963684e.png)